Assets/Scripts/SS3D/Systems/Tile/TileMap.cs

```diff
@@ -421,5 +455,35 @@ private void UpdateAllAdjacencies()
                 }
             }
         }
+
+        /// <summary>
+        /// Clear items in the scene that could conflict with loaded items
+        /// </summary>
+        private void ClearAllItemsInScene()
+        {
+            // Find all Item components in the scene
+            var allItems = GameObject.FindObjectsOfType<SS3D.Systems.Inventory.Items.Item>();
+            
+            foreach (var item in allItems)
+            {
+                // Skip items that are in containers (player inventory, etc.)
+                if (item.Container != null)
+                {
+                    continue;
+                }
+                
+                // Skip items that already have PlacedItemObject (they're already tracked)
+                if (item.GetComponent<PlacedItemObject>() != null)
+                {
+                    continue;
+                }
+                
+                // Destroy items that are in the world but not tracked by TileMap
+                if (item.gameObject != null)
+                {
+                    DestroyImmediate(item.gameObject);
+                }
+            }
```

Comment:
```suggestion
            Item[] allItems = FindObjectsOfType<Item>();
            
            foreach (Item item in allItems)
            {
                // Skip items that are in containers (player inventory, etc.)
                // and items that already have PlacedItemObject (they're already tracked)
                if (item.Container != null || item.GetComponent<PlacedItemObject>() != null)
                {
                    continue;
                }
                
                // Destroy items that are in the world but not tracked by TileMap
                if (item.gameObject != null)
                {
                    DestroyImmediate(item.gameObject);
                }
            }
```
If all items are either in containers or being tracked (as i understand), what is this function supposed to clear?

Reply:
This function is mainly a defensive cleanup to remove any world items that aren't in containers and aren't tracked by the TileMap. Ideally, this function would do nothing, but it is mainly there to prevent issues from stray or untracked items (eg. items that were placed in the scene by the Unity Editor (not registered with the TileMap at runtime)). Let me know if this doesn't make any sense

Reply:
Then it should log a warning about what items are being leftover. Also change the description and name to represent that the function deletes only items out of containers and without a tracker